src/fhir.js

```diff
@@ -384,21 +384,35 @@ class Patient extends FHIRObject {
     }
   }
 
-  findRecords(profile) {
-    const classInfo = this._modelInfo.findClass(profile);
-    if (classInfo == null) {
-      console.error(`Failed to find type info for ${profile}`);
-      return [];
+  findRecords(profile, retrievedDetails) {
+    let classInfo;
+    if (retrievedDetails) {
+      classInfo = this._modelInfo.findClass(retrievedDetails.dataType);
```

Comment:
This will always be `undefined`, as `datatype` is not camelCase in the [cql-execution code](https://github.com/cqframework/cql-execution/blob/0d0ab598ce17ed61775bc4eb058654da81f74df6/src/types/cql-patient.interfaces.ts#L30)